### ISSUE TITLE
Want Per-user onionname with clientAuth protection

### ISSUE THREAD

**Issue by awtbrain** [blueprint-freespeech/ricochet-refresh#111]
1. Normal mode:

A: hey B add me my id is `rr://ooooooooooonionidhere`
B: sure
A hey C add me my id is `rr://ooooooooooonionidhere`
C: sure
B: I hate A. I gonna dox A's id on the nakedweb
D: adding A's id
A: I know B fucked me but **I cannot delete my ID because it is already used by other firends**

2. Paranoid mode:

A: hey B add me my id is `rr://ooooooooooonionidhere?psk=OnioNAuthKeyhere`
B: sure
A hey C add me my id is `rr://nnneexxxttonionidhere?psk=welpwelpwelp`
C: sure
B: I hate A. I gonna dox A's id on the nakedweb
D: adding A's id
A: I know B fucked me and I'm going to delete `ooooooooooonionidhere` now! **Other contacts are not affected**

tldr;
1. Per-contact per-onion.
2. Use OnionService's auth mechanizm for enhanced privacy (to connect `ooooooooooonionidhere` you need to know the key)

**Comment by pospeselr** [comment]
Related (believe it or not) to ongoing work to fix #73

**Comment by pospeselr** [comment]
This will be resolved once gosling is used for user authentication. See: https://github.com/blueprint-freespeech/gosling/blob/main/docs/protocol.md
